# Exercise 3 - Confidence Interval for Bias and Regression Coefficient

# Objective : Learning the construction of Confidence Intervals for $\beta_0$ and $\beta_1$

# Constuct the confidence intervals with confidence coefficient 0.97 for the bias and regression coefficient using the TV Sales Data (Amount Spend on TV advertisement and the level of Sales achieved)

# Step 1 : Importing the data and slicing response and regressor columns

In [5]:
import pandas as pd
df=pd.read_csv("sales.csv")
X=df[["TV"]].values
y=df[["SALES"]].values

# Step 2 : Fitting the model and obtaining the estimates of bias and regression coefficient

In [28]:
from sklearn.linear_model import LinearRegression
model=LinearRegression()
model.fit(X,y)
b0_hat=model.intercept_[0]
b1_hat=model.coef_[0,0]

0.055289582581731216

# Step 3 :  Estimating the error variance

In [7]:
est_var=sum((model.predict(X)-y)*(model.predict(X)-y))/(len(X)-2)
est_var=est_var[0]

# Step 4 : Obtaining the quantile (t-table) values

In [8]:
from scipy.stats import t

In [9]:
# t.ppf(0.90,13) This call gives the t-table value where the left tail area is 0.90 and the right tail area
# is 0.10 # this is compable with qt in R

In [10]:
conf_coef=0.97
alpha=(1-conf_coef)
t_value=t.ppf((1-(alpha/2)),len(X)-2)
t_value

2.3132309601308547

# Step 5 : Computing Standard Errors

In [18]:
x_bar=np.mean(X)
Sxx=sum((X-x_bar)*(X-x_bar))[0]
print(Sxx)

183723.0944


In [22]:
import numpy as np
se_b1=np.sqrt(est_var/Sxx)
fact=(1/len(X))+(x_bar*x_bar/Sxx)
se_b0=np.sqrt(est_var*fact)
print(se_b0,se_b1)

0.6917389463795325 0.004543191124991639


# Step 6 : Computing Confidence limits

In [24]:
l_b0=b0_hat-t_value*se_b0
u_b0=b0_hat+t_value*se_b0
l_b1=b1_hat-t_value*se_b1
u_b1=b1_hat+t_value*se_b1

# Step 7 : Printing the confidence intervals 

In [27]:
print("Confidence interval for b_0", "(",l_b0,",",u_b0,")")
print("Confidence interval for b_1", "(",l_b1,",",u_b1,")")

Confidence interval for b_0 ( 5.838649297482166 , 9.03895319166903 )
Confidence interval for b_1 ( 0.044780132213608825 , 0.0657990329498536 )


# Step 8 : Confidence interval with confidence coefficient 99%

In [29]:
conf_coef=0.99
alpha=(1-conf_coef)
t_value=t.ppf((1-(alpha/2)),len(X)-2)
l_b0=b0_hat-t_value*se_b0
u_b0=b0_hat+t_value*se_b0
l_b1=b1_hat-t_value*se_b1
u_b1=b1_hat+t_value*se_b1
print("Confidence interval for b_0", "(",l_b0,",",u_b0,")")
print("Confidence interval for b_1", "(",l_b1,",",u_b1,")")

Confidence interval for b_0 ( 5.496857816552588 , 9.380744672598608 )
Confidence interval for b_1 ( 0.04253532001836627 , 0.06804384514509616 )


# Conclusion 

# The confidence intervals with confidence coefficient 0.97 for the bias and regression coefficient is 

## Confidence interval for $\beta_0$ ( 5.496857816552588 , 9.380744672598608 )
## Confidence interval for $\beta_1$ ( 0.04253532001836627 , 0.06804384514509616 )